presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1348,11 +1356,27 @@ else if (expression instanceof LongLiteral) {
                     }
 
                     // otherwise, just use the expression as is
+                    boolean orderByOnExpression = false;
                     if (orderByExpression == null) {
                         orderByExpression = expression;
+                        orderByOnExpression = true;
                     }
 
                     ExpressionAnalysis expressionAnalysis = analyzeExpression(orderByExpression, sourceScope);
+                    if (orderByOnExpression) {
+                        // Try to resolve within orderByScope. If it is possible, then we warn - standard behavior differs from legacy one.
+                        for (Expression columnExpression : expressionAnalysis.getColumnReferences()) {
+                            Optional<ResolvedField> resolvedField = orderByScope.tryResolveField(columnExpression);
```

Comment:
I see, this makes sense. I'll wait till #7354 is in. So in addition to this code I need to check if resolved field column RelationId references current table, but not other (basically, from subquery)?

Other approach I was thinking is to run 
```
ExpressionAnalysis expressionAnalysis = analyzeExpression(
  orderByExpression,
  sourceScope);
// .. and run with orderByScope
ExpressionAnalysis expressionAnalysisNew = analyzeExpression(
  orderByExpression, 
  orderByScope);

// Check column references are the same in expressionAnalysis and 
// expressionAnalysisNew by equal compare.
```
Seems like a very good approach, which only requires equality from referenced columns and treats process of resolution completely as blackbox - just results need to match.